File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy=new ListNode(0);
        dummy.next=head;
        int length=length(head);
        int removenode=length-n;
        ListNode curr=dummy;
        if(head==null || head.next==null) return null;
        for(int i=0;i<removenode;i++){
            curr=curr.next;
        }
        curr.next=curr.next.next;
        return dummy.next;
    }
    public int length(ListNode head){
        int c=0;
        ListNode curr=head;
        while(curr!=null){
            c+=1;
            curr=curr.next;
        }
        return c;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        llength=self.length(head)
        end=llength-n-1
        if(end==-1):
            return head.next
        i=0
        curr=head
        while(i<end):
            curr=curr.next
            i+=1
        curr.next=curr.next.next
        return head
        
    
    
    
    def length(self,head):
        c=0
        if head is None:
            return c
        curr=head
        while curr is not None:
            c+=1
            curr=curr.next
        return c